### ISSUE TITLE
Feature Request: Automaticly Detect the denomination of minegeld notes in deposits

### ISSUE THREAD

**Issue by ApolloX939** [Tmanyo/bank_accounts#4]
This means instead of code for 3 separate "slots" we have 3 slots that will auto detect what the value of the stack is...

I.E. Currently you can only deposit 1s 5s and 10s... with purposed feature we can support all of currency's items. (From 5 cents to 100 minegeld)

Another Example with the feature in place:

I can place my money for deposit how ever I want... it won't matter because the code will tell oh the first is 10 the second is 1 and the third is .25 and not dupe a single cent or minegeld.

Now this will mean code wise you'd need to separate/branch off the functions to process each denomination and simply do as the pseudo code below would show for deposits

```Lua
if stack:get_name() == "currency:minegeld_cent_5" then
    -- We have our function call to process how many are in the stack and add them to the account balance
elseif stack:get_name() == "currency:minegeld_cent_10" then
    -- Same deal as above just for 10 cents instead
elseif stack:get_name() == "currency:minegeld_cent_25" then
   -- I think you get the point for the deposit logic
```

What this pseudo code allows us to do is this... auto detect based on what is there in that slot. (Who cares if it's in a different order)

**Comment by ApolloX939** [comment]
Actually at least the way my PR does it, makes it look strange... I.E. I am depositing multiple currency:minegeld_5...

But in my example I placed the notes into the ones slot... instead of waiting for me to hit a button it takes it and processes it. (So it looks like it just disappears)

I think I made the ATM act just like reality... you go up, select to deposit cash, and the machine will promptly grab the cash quickly out of your hands.
